go/kv/internal.go

```diff
@@ -65,3 +67,9 @@ func (this *internalKVStore) GetKeyValue(key string) (value string, found bool,
 func (this *internalKVStore) DistributePairs(kvPairs [](*KVPair)) (err error) {
 	return nil
 }
+
+func (this *internalKVStore) DeleteRecursive(key string) (err error) {
+	query := fmt.Sprintf("DELETE FROM kv_store WHERE store_key LIKE '%s%%'", key)
```

Comment:
This will delete `/mysql/master/important` when we forget the `import` cluster, seeing that `import` is a substring of `important`.
If we forget a cluster called `po`, this will forget `import`, `important`, `poo`, `topology` and any other cluster which has `po` as substring.

Reply:
Yes, you are right with the first case, but the second won't be true, as the key is already in the form {kv_prefix}/key.
Think changing it to:
`query := fmt.Sprintf("DELETE FROM kv_store WHERE store_key = '%s' OR store_key LIKE '%s/%%'", key, key)`
should solve the issue.